chain/messagepool/messagepool.go

```diff
@@ -135,9 +135,9 @@ type MessagePool struct {
 	localAddrs map[address.Address]struct{}
 
 	// do NOT access this map directly, use getPendingMset, setPendingMset, deletePendingMset, forEachPending, and clearPending respectively
-	pending map[address.Address]*msgSet
+	pending sync.Map //threadsafe map[address.Address]*msgSet
 
-	keyCache map[address.Address]address.Address
+	keyCache sync.Map // map[address.Address]address.Address
```

Comment:
I think the improvement between this PR and not having it is very large, and the improvement between map.Sync and xsync is much much smaller. If we can first land this PR, we can def discuss the idea of using a non standard synchronized map, but already we are getting yuuuge perf improvements so it may be moot